src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala

```diff
@@ -268,9 +268,15 @@ private[internal] trait TypeConstraints {
 
         val newInst =
           if (up || tvar.constr.hiBounds.exists(isSingleType)) { // If we have a singleton upper bound then we should use it.
-            if (depth.isAnyDepth) glb(tvar.constr.hiBounds) else glb(tvar.constr.hiBounds, depth)
+            val hiBounds = tvar.constr.hiBounds
+            if (hiBounds.isEmpty) genPolyType(cloneSymbols(tvar.typeParams), AnyTpe)
+            else if (depth.isAnyDepth) glb(hiBounds)
+            else glb(hiBounds, depth)
           } else {
-            if (depth.isAnyDepth) lub(tvar.constr.loBounds) else lub(tvar.constr.loBounds, depth)
+            val loBounds = tvar.constr.loBounds
+            if (loBounds.isEmpty) genPolyType(cloneSymbols(tvar.typeParams), NothingTpe)
+            else if (depth.isAnyDepth) lub(loBounds)
+            else lub(loBounds, depth)
```

Comment:
I'm not sure about these changes. The other option would be to special-case `Any` and `Nothing` here:
https://github.com/scala/scala/blob/d8d63b2da0d62db3c8239ef2886adf134ae7269b/src/reflect/scala/reflect/internal/tpe/TypeComparers.scala#L384-L402

Reply:
It is kind of weird that usually Any/Nothing are kind-polymorphic, but not when comparing to TypeVars. If you have time to add the special case in isSubHKTypeVar and drop the PolyType around Any/Nothing here, I think that would be neater.